### ISSUE TITLE
增强 constant folding pass

### ISSUE THREAD

**Issue by wwens7** [BD-Seed-HHW/xpu_graph#327]
### Describe the feature🚀

背景：当前constant folding不彻底，并且没有合并相同的constant，导致性能和显存都有一定的提升空间（from @ zhangjihang）

feature 拆解：
1. 支持已折叠常量的显存释放
2. 支持相同常量的合并

### 已折叠常量显存释放
当前执行图优化后，常量折叠会将可以预先计算好的表达式在编译期进行计算，减少 runtime overhead。如，一次 gemm 操作的其中一个操作数是常量 （weight），且这个 weight 需要被 relu 两次，则：relu -> relu -> mm 会被折叠成 get_attr('_xpugraph_folded_relu_1_1') -> mm，原始的 weight 节点，两个 relu 节点都会从图中删去

虽然节点被删除，但优化后图 gm 中的 state_dict 依然包含了原始的 self____buffers__const_weight，甚至还会包含一份 _xpugraph_folded_relu_0，即虽然图被简化了，但原始的权重数据、折叠中产生的权重数据仍然占用 Global Memory。

综上，可以设计一个 Constant Pruning Pass 来解决这个问题，它应该在：
1. 所常量折叠 Pass 之后；
2. 所有常量合并 Pass 之后（如果有）
3. 所有DCE Pass 之后
执行。另外，该 Pass 的目标不应该是修改图，而是修改 GraphModule 对象本身，即删除它上面不再需要的属性（parameters 和 buffers），从而触发 Python 垃圾回收器释放显存。

初步拟定三步来执行：
1. 收集优化后 gm.graph.nodes 中所有 'get_attr' -> set 1
2. 收集 gm.state_dict().keys() -> set 2
3. 根据 set 1、2，收集不再被需要的 const
4. Pruning

concern：gm.graph.nodes 中的 'get_attr' 与 gm.state_dict().keys()名称格式有出入（如，PyTorch 会自动添加前缀和替换下划线等等）

### References

-

### Environment details

-

**Comment by zhangjihang-BD** [comment]
还需要做一件事，就是折叠一些ones_like & full_like

**Comment by wwens7** [comment]
> 还需要做一件事，就是折叠一些ones_like & full_like

get
